

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>ontmont package &#8212; ontmont 0.0.4 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=3ee479438cf8b5e0d341" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=3ee479438cf8b5e0d341" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=3ee479438cf8b5e0d341"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ontmont';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ontmont" href="modules.html" />
    <link rel="prev" title="Welcome to ontmont’s documentation!" href="index.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search the docs ..."
         aria-label="Search the docs ..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">

  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
  
    <p class="title logo__title">ontmont 0.0.4 documentation</p>
  
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav class="navbar-nav">
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item pst-header-nav-item">
  <a class="nav-link nav-internal" href="#">
    ontmont package
  </a>
</li>


<li class="nav-item pst-header-nav-item current active">
  <a class="nav-link nav-internal" href="modules.html">
    ontmont
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

 <script>
 document.write(`
   <button class="btn navbar-btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
        </div>
      
      
        <div class="navbar-item">

<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

 <script>
 document.write(`
   <button class="btn navbar-btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
    </div>
  

  
    <button class="sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav class="navbar-nav">
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item pst-header-nav-item">
  <a class="nav-link nav-internal" href="#">
    ontmont package
  </a>
</li>


<li class="nav-item pst-header-nav-item current active">
  <a class="nav-link nav-internal" href="modules.html">
    ontmont
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">

<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
<nav class="bd-docs-nav bd-links"
     aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">Section Navigation</p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">ontmont package</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item">



<nav aria-label="Breadcrumb" class="d-print-none">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a href="index.html" class="nav-link" aria-label="Home">
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
    <li class="breadcrumb-item active" aria-current="page">ontmont package</li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="ontmont-package">
<h1>ontmont package<a class="headerlink" href="#ontmont-package" title="Permalink to this heading">#</a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this heading">#</a></h2>
</section>
<section id="module-ontmont.bundle">
<span id="ontmont-bundle-module"></span><h2>ontmont.bundle module<a class="headerlink" href="#module-ontmont.bundle" title="Permalink to this heading">#</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ontmont.bundle.make_brk_supports">
<span class="sig-prename descclassname"><span class="pre">ontmont.bundle.</span></span><span class="sig-name descname"><span class="pre">make_brk_supports</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/bundle.py#L174-L190"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.bundle.make_brk_supports" title="Permalink to this definition">#</a></dt>
<dd><p>Count supports for unique breakpoint coordinates</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>bundle</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pandas.Series</strong> – Count of unique breakpoints, taking chrom, pos, ori into account</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.bundle.make_brk_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.bundle.</span></span><span class="sig-name descname"><span class="pre">make_brk_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">brk_supports</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unilateral_score_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bilateral_score_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/bundle.py#L66-L133"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.bundle.make_brk_table" title="Permalink to this definition">#</a></dt>
<dd><p>Create a dataframe of breakpoints</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bundle</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> variables</p></li>
<li><p><strong>brk_supports</strong> (<em>dict</em><em> | </em><em>pandas.Series</em>) – Number of support for each breakpoint coordinate</p></li>
<li><p><strong>unilateral_score_cutoff</strong> (<em>int</em><em>, </em><em>optional</em>) – Alignment score cutoff for unilateral IR. Defaults to 5.</p></li>
<li><p><strong>bilateral_score_cutoff</strong> (<em>int</em><em>, </em><em>optional</em>) – Alignment score cutoff for bilateral IR. Defaults to 8.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pandas.DataFrame</strong> – table of breakpoint coordinates with support and IR statistics</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.bundle.make_brks_bundle">
<span class="sig-prename descclassname"><span class="pre">ontmont.bundle.</span></span><span class="sig-name descname"><span class="pre">make_brks_bundle</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reads_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw_palindrome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw_holliday</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[15,</span> <span class="pre">30,</span> <span class="pre">60]</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/bundle.py#L136-L171"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.bundle.make_brks_bundle" title="Permalink to this definition">#</a></dt>
<dd><p>Make a list of <code class="docutils literal notranslate"><span class="pre">BreapointChain</span></code> based on alignment table, genome, and alignment parameters</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>reads_df</strong> (<em>pandas.DataFrame</em>) – Table of read alignment statistics</p></li>
<li><p><strong>genome</strong> (<em>pyfaidx.Fasta</em>) – Genome fasta</p></li>
<li><p><strong>sw_palindrome</strong> (<em>swalign.LocalAlignment</em>) – Parameters for detecting IR</p></li>
<li><p><strong>sw_holliday</strong> (<em>swalign.LocalAlignment</em>) – Parameters for detecting homology</p></li>
<li><p><strong>margins</strong> (<em>list</em><em>, </em><em>optional</em>) – Bases to slice from breakpoints. Defaults to [15, 30, 60].</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.bundle.make_seg_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.bundle.</span></span><span class="sig-name descname"><span class="pre">make_seg_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seg_supports</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">segment_score_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/bundle.py#L9-L63"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.bundle.make_seg_table" title="Permalink to this definition">#</a></dt>
<dd><p>Create a dataframe based on a <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> bundle and supports dict</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bundle</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> variables</p></li>
<li><p><strong>seg_supports</strong> (<em>dict</em><em> | </em><em>pandas.Series</em>) – Number of support for each breakpoint coordinate</p></li>
<li><p><strong>segment_score_cutoff</strong> (<em>int</em><em>, </em><em>optional</em>) – Alignment score cutoff the IR found in the segment. Defaults to 5.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pandas.DataFrame</strong> – table of segments coordinate with supports and IR statistics</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.bundle.make_tra_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.bundle.</span></span><span class="sig-name descname"><span class="pre">make_tra_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tra_supports</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/bundle.py#L193-L236"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.bundle.make_tra_table" title="Permalink to this definition">#</a></dt>
<dd><p>Make a table of SVs based on bundle and number of supports</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bundle</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p></li>
<li><p><strong>tra_supports</strong> (<em>dict</em><em> | </em><em>pandas.Series</em>) – Support count for SVs</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pandas.DataFrame</strong> – Table of SVs, duplicate removed, removed if located on contig termini</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ontmont.cli">
<span id="ontmont-cli-module"></span><h2>ontmont.cli module<a class="headerlink" href="#module-ontmont.cli" title="Permalink to this heading">#</a></h2>
</section>
<section id="module-ontmont.collect">
<span id="ontmont-collect-module"></span><h2>ontmont.collect module<a class="headerlink" href="#module-ontmont.collect" title="Permalink to this heading">#</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.extract_split_alignments">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">extract_split_alignments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reads</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_reads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L10-L40"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.extract_split_alignments" title="Permalink to this definition">#</a></dt>
<dd><p>Extract SplitAlignment objects from IteratorRow with a max_reads parameter</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>reads</strong> (<em>pysam.IteratorRow</em>) – Reads fetched from a pysam.Alignmentfile</p></li>
<li><p><strong>max_reads</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of reads to extract at maximum. Defaults to 500.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – list of <code class="docutils literal notranslate"><span class="pre">SplitAlignment</span></code> objects</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.find_presence_of_matching_sv">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">find_presence_of_matching_sv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L272-L297"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.find_presence_of_matching_sv" title="Permalink to this definition">#</a></dt>
<dd><p>Check overlap of sv2 for sv1 table</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sv1</strong> (<em>pandas.DataFrame</em>) – SV table to label matching SVs</p></li>
<li><p><strong>sv2</strong> (<em>pandas.DataFrame</em>) – SV table reference to check presence of overlap</p></li>
<li><p><strong>margin</strong> (<em>int</em><em>, </em><em>optional</em>) – Margin (bp) of breakpoint coordinate difference. Defaults to 50.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pd.Series</strong> – <cite>{True, False}</cite> list of matches. Length equal to sv1 row size.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.fix_lower_support_coordinates">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">fix_lower_support_coordinates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">complexes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">coord_map</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L152-L169"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.fix_lower_support_coordinates" title="Permalink to this definition">#</a></dt>
<dd><p>Map breakpoint of lower support to close-by breakpoint with higher support</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>complexes</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p></li>
<li><p><strong>coord_map</strong> (<em>dict</em>) – Map of str(<code class="docutils literal notranslate"><span class="pre">Breakpoint</span></code>) coordinates</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code>, mapped to fixed coordinates</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.get_breakpoint_support_from_bundle">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">get_breakpoint_support_from_bundle</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">complexes</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L102-L115"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.get_breakpoint_support_from_bundle" title="Permalink to this definition">#</a></dt>
<dd><p>Get breakpoint support count</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>complexes</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>collections.Counter</strong> – Support for str(<code class="docutils literal notranslate"><span class="pre">Breakpoint</span></code>) coordinates</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.get_normalized_sv">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">get_normalized_sv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tra</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L359-L375"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.get_normalized_sv" title="Permalink to this definition">#</a></dt>
<dd><p>Sort (normalize) a <code class="docutils literal notranslate"><span class="pre">BreakpointPair</span></code></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>tra</strong> (<code class="docutils literal notranslate"><span class="pre">BreakpointPair</span></code>) – Pair of breakpoints</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – Sorted breakpoint coordinates, flattened</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.get_svtype">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">get_svtype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tra</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L202-L238"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.get_svtype" title="Permalink to this definition">#</a></dt>
<dd><p>Get SV type string for a given <cite>BreakpointPair</cite></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>tra</strong> (<cite>BreakpointPair</cite>) – Paired breakpoint object</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If no SV type has been assigned</p>
</dd>
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>str</strong> – SV type string</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.make_brks_bundle">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">make_brks_bundle</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reads_df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L84-L100"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.make_brks_bundle" title="Permalink to this definition">#</a></dt>
<dd><p>Make a list of <code class="docutils literal notranslate"><span class="pre">BreapointChain</span></code> based on alignment table</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>reads_df</strong> (<em>pandas.DataFrame</em>) – Table of read alignment statistics</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.make_tumor_sv_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">make_tumor_sv_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">complexes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_support</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L324-L357"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.make_tumor_sv_table" title="Permalink to this definition">#</a></dt>
<dd><p>Make SV table from list of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>complexes</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p></li>
<li><p><strong>sv</strong> (<em>pandas.DataFrame</em><em>, </em><em>optional</em>) – Table of source SVs as reference for <cite>in_source</cite> flag. Defaults to None</p></li>
<li><p><strong>margin</strong> (<em>int</em><em>, </em><em>optional</em>) – Margin (bp) for merging clustered breakpoints. Defaults to 10.</p></li>
<li><p><strong>get_support</strong> (<em>bool</em><em>, </em><em>optional</em>) – Merge breakpoints with same coordinates and add count as <cite>support</cite>. Defaults to True.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pandas.DataFrame</strong> – SV table from bundle [, with <cite>in_source</cite> labels] [, collapsed by coordinate with support counts]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.map_similar_coordinate_to_higher_rank">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">map_similar_coordinate_to_higher_rank</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">complexes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">breakpoint_support</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L117-L150"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.map_similar_coordinate_to_higher_rank" title="Permalink to this definition">#</a></dt>
<dd><p>Make mapping of close-by coordinates, with breakpoints of higher support taking priority</p>
<dl class="field-list">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>complexes</strong> (<em>list</em>) – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p></li>
<li><p><strong>breakpoint_support</strong> (<em>dict</em><em> | </em><em>collections.Counter</em>) – Support for breakpoint coordinates</p></li>
<li><p><strong>margin</strong> (<em>int</em><em>, </em><em>optional</em>) – Margin (bp) to merge close-by coordinates. Defaults to 10.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>tuple</strong> – tuple containing:</p>
<p>coord_map (dict): src -&gt; dst coordinate</p>
<p>coord_map_log (tuple): (max_coord, src_count, max_count) [only for debugging]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.normalize_sv_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">normalize_sv_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom1_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'chromosome_1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom2_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'chromosome_2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos1_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'position_1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos2_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'position_2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ori1_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'strand_1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ori2_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'strand_2'</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L171-L200"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.normalize_sv_table" title="Permalink to this definition">#</a></dt>
<dd><p>Sort breakpoint1 and breakpoint2 of a SV table</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sv</strong> (<em>pandas.DataFrame</em>) – Table of SVs</p></li>
<li><p><strong>chrom1_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Defaults to ‘chromosome_1’.</p></li>
<li><p><strong>chrom2_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Defaults to ‘chromosome_2’.</p></li>
<li><p><strong>pos1_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Defaults to ‘position_1’.</p></li>
<li><p><strong>pos2_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Defaults to ‘position_2’.</p></li>
<li><p><strong>ori1_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Defaults to ‘strand_1’.</p></li>
<li><p><strong>ori2_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Defaults to ‘strand_2’.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pandas.DataFrame</strong> – Sorted (normalized) SV table</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.pull_breakpoints_from_bam_files">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">pull_breakpoints_from_bam_files</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam_paths</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_read_table</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L299-L322"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.pull_breakpoints_from_bam_files" title="Permalink to this definition">#</a></dt>
<dd><p>Get <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> list from BAM file according to an input SV table</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bam_paths</strong> (<em>pysam.AlignmentFile</em>) – BAM file</p></li>
<li><p><strong>sv</strong> (<em>pandas.DataFrame</em>) – SV table</p></li>
<li><p><strong>get_read_table</strong> (<em>bool</em><em>, </em><em>optional</em>) – Return read table as well. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list [, pandas.DataFrame]</strong> – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> [, table of read alignment stats]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.pull_breakpoints_from_reads_in_sv_regions">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">pull_breakpoints_from_reads_in_sv_regions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tra</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_read_table</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_n_breakpoint</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L42-L82"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.pull_breakpoints_from_reads_in_sv_regions" title="Permalink to this definition">#</a></dt>
<dd><p>Extract and append <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> objects from a bam file and a table of SVs</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bam</strong> (<em>pysam.AlignmentFile</em>) – BAM file</p></li>
<li><p><strong>tra</strong> (<em>pandas.DataFrame</em>) – Table of SVs</p></li>
<li><p><strong>get_read_table</strong> (<em>bool</em><em>, </em><em>optional</em>) – Return table of read alignment stats. Defaults to False.</p></li>
<li><p><strong>min_n_breakpoint</strong> (<em>int</em><em>, </em><em>optional</em>) – Minimum number of breakpoints required to be saved. Useful in selecting complex rearrangements if the number is high. Defaults to 3.</p></li>
<li><p><strong>margin</strong> (<em>int</em><em>, </em><em>optional</em>) – Margin (bp) from breakpoints to fetch reads. Defaults to 10.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – List of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.collect.pull_sv_supporting_reads_from_bundle">
<span class="sig-prename descclassname"><span class="pre">ontmont.collect.</span></span><span class="sig-name descname"><span class="pre">pull_sv_supporting_reads_from_bundle</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/collect.py#L240-L270"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.collect.pull_sv_supporting_reads_from_bundle" title="Permalink to this definition">#</a></dt>
<dd><p>Filter bundle to include <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code> objects that have breakpoints matching that of the input sv table</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sv</strong> (<em>pandas.DataFrame</em>) – SV table</p></li>
<li><p><strong>bundle</strong> (<em>list</em>) – list of <code class="docutils literal notranslate"><span class="pre">BreapointChain</span></code></p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>list</strong> – Filtered list of <code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ontmont.datatypes">
<span id="ontmont-datatypes-module"></span><h2>ontmont.datatypes module<a class="headerlink" href="#module-ontmont.datatypes" title="Permalink to this heading">#</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.Breakpoint">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">Breakpoint</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orientation</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L203-L236"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Breakpoint" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ontmont.datatypes.Breakpoint.chroms">
<span class="sig-name descname"><span class="pre">chroms</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['1',</span> <span class="pre">'2',</span> <span class="pre">'3',</span> <span class="pre">'4',</span> <span class="pre">'5',</span> <span class="pre">'6',</span> <span class="pre">'7',</span> <span class="pre">'8',</span> <span class="pre">'9',</span> <span class="pre">'10',</span> <span class="pre">'11',</span> <span class="pre">'12',</span> <span class="pre">'13',</span> <span class="pre">'14',</span> <span class="pre">'15',</span> <span class="pre">'16',</span> <span class="pre">'17',</span> <span class="pre">'18',</span> <span class="pre">'19',</span> <span class="pre">'20',</span> <span class="pre">'21',</span> <span class="pre">'22',</span> <span class="pre">'X',</span> <span class="pre">'Y',</span> <span class="pre">'M',</span> <span class="pre">'chr1',</span> <span class="pre">'chr2',</span> <span class="pre">'chr3',</span> <span class="pre">'chr4',</span> <span class="pre">'chr5',</span> <span class="pre">'chr6',</span> <span class="pre">'chr7',</span> <span class="pre">'chr8',</span> <span class="pre">'chr9',</span> <span class="pre">'chr10',</span> <span class="pre">'chr11',</span> <span class="pre">'chr12',</span> <span class="pre">'chr13',</span> <span class="pre">'chr14',</span> <span class="pre">'chr15',</span> <span class="pre">'chr16',</span> <span class="pre">'chr17',</span> <span class="pre">'chr18',</span> <span class="pre">'chr19',</span> <span class="pre">'chr20',</span> <span class="pre">'chr21',</span> <span class="pre">'chr22',</span> <span class="pre">'chrX',</span> <span class="pre">'chrY',</span> <span class="pre">'chrM']</span></em><a class="headerlink" href="#ontmont.datatypes.Breakpoint.chroms" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.Breakpoint.get_breakpoint_seqs">
<span class="sig-name descname"><span class="pre">get_breakpoint_seqs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">margin</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L215-L224"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Breakpoint.get_breakpoint_seqs" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.BreakpointChain">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">BreakpointChain</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">brks_iterable</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L62-L88"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.BreakpointChain" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.BreakpointChain.get_segments">
<span class="sig-name descname"><span class="pre">get_segments</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L82-L88"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.BreakpointChain.get_segments" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.BreakpointChain.get_transitions">
<span class="sig-name descname"><span class="pre">get_transitions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sort_transition</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L69-L79"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.BreakpointChain.get_transitions" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.BreakpointPair">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">BreakpointPair</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">brk1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">brk2</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L239-L245"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.BreakpointPair" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.Sample">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">Sample</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">long_sample</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L91-L118"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Sample" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ontmont.datatypes.Sample.long_samples">
<span class="sig-name descname"><span class="pre">long_samples</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['14472B_201',</span> <span class="pre">'14472B_202',</span> <span class="pre">'14472B_500',</span> <span class="pre">'14472B_501',</span> <span class="pre">'14472C_203',</span> <span class="pre">'14472C_204',</span> <span class="pre">'14472C_502',</span> <span class="pre">'14472D_101',</span> <span class="pre">'14472D_102',</span> <span class="pre">'14472D_103',</span> <span class="pre">'14472D_104',</span> <span class="pre">'14472D_205',</span> <span class="pre">'14472_100',</span> <span class="pre">'14472_300']</span></em><a class="headerlink" href="#ontmont.datatypes.Sample.long_samples" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ontmont.datatypes.Sample.sample2group">
<span class="sig-name descname"><span class="pre">sample2group</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'14472B_201':</span> <span class="pre">'2',</span> <span class="pre">'14472B_202':</span> <span class="pre">'2',</span> <span class="pre">'14472B_500':</span> <span class="pre">'5',</span> <span class="pre">'14472B_501':</span> <span class="pre">'5',</span> <span class="pre">'14472C_203':</span> <span class="pre">'2',</span> <span class="pre">'14472C_204':</span> <span class="pre">'2',</span> <span class="pre">'14472C_502':</span> <span class="pre">'5',</span> <span class="pre">'14472D_101':</span> <span class="pre">'1',</span> <span class="pre">'14472D_102':</span> <span class="pre">'1',</span> <span class="pre">'14472D_103':</span> <span class="pre">'1',</span> <span class="pre">'14472D_104':</span> <span class="pre">'1',</span> <span class="pre">'14472D_205':</span> <span class="pre">'2',</span> <span class="pre">'14472_100':</span> <span class="pre">'1',</span> <span class="pre">'14472_300':</span> <span class="pre">'3'}</span></em><a class="headerlink" href="#ontmont.datatypes.Sample.sample2group" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ontmont.datatypes.Sample.sample2short">
<span class="sig-name descname"><span class="pre">sample2short</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'14472B_201':</span> <span class="pre">'201',</span> <span class="pre">'14472B_202':</span> <span class="pre">'202',</span> <span class="pre">'14472B_500':</span> <span class="pre">'500',</span> <span class="pre">'14472B_501':</span> <span class="pre">'501',</span> <span class="pre">'14472C_203':</span> <span class="pre">'203',</span> <span class="pre">'14472C_204':</span> <span class="pre">'204',</span> <span class="pre">'14472C_502':</span> <span class="pre">'502',</span> <span class="pre">'14472D_101':</span> <span class="pre">'101',</span> <span class="pre">'14472D_102':</span> <span class="pre">'102',</span> <span class="pre">'14472D_103':</span> <span class="pre">'103',</span> <span class="pre">'14472D_104':</span> <span class="pre">'104',</span> <span class="pre">'14472D_205':</span> <span class="pre">'205',</span> <span class="pre">'14472_100':</span> <span class="pre">'100',</span> <span class="pre">'14472_300':</span> <span class="pre">'300'}</span></em><a class="headerlink" href="#ontmont.datatypes.Sample.sample2short" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.Segments">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">Segments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L154-L168"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Segments" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.Segments.get_list">
<span class="sig-name descname"><span class="pre">get_list</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L160-L168"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Segments.get_list" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.SplitAlignment">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">SplitAlignment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cigarstring</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">refname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_pos</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strand</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L6-L59"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.SplitAlignment" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.SplitAlignment.extract_cigar_field">
<span class="sig-name descname"><span class="pre">extract_cigar_field</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L20-L43"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.SplitAlignment.extract_cigar_field" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.SplitAlignment.get_cigar_tuples">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">get_cigar_tuples</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cigarstring</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L45-L59"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.SplitAlignment.get_cigar_tuples" title="Permalink to this definition">#</a></dt>
<dd><p>Returns a cigar tuple from a CIGAR string</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ontmont.datatypes.Transitions">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">Transitions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L121-L151"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Transitions" title="Permalink to this definition">#</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ontmont.datatypes.Transitions.get_list">
<span class="sig-name descname"><span class="pre">get_list</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L127-L151"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.Transitions.get_list" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.datatypes.get_breakpoint_seqs">
<span class="sig-prename descclassname"><span class="pre">ontmont.datatypes.</span></span><span class="sig-name descname"><span class="pre">get_breakpoint_seqs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/datatypes.py#L171-L200"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.datatypes.get_breakpoint_seqs" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</section>
<section id="module-ontmont.irs">
<span id="ontmont-irs-module"></span><h2>ontmont.irs module<a class="headerlink" href="#module-ontmont.irs" title="Permalink to this heading">#</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.align_two_sequences">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">align_two_sequences</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L14-L31"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.align_two_sequences" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.calc_distance_score">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">calc_distance_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dist1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L9-L12"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.calc_distance_score" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.calc_pval_bilateral_ir">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">calc_pval_bilateral_ir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">src_score</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_iter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L120-L141"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.calc_pval_bilateral_ir" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.calc_pval_holliday">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">calc_pval_holliday</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">src_score</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_iter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L254-L274"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.calc_pval_holliday" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.calc_pval_onesided_ir">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">calc_pval_onesided_ir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">direction</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">src_score</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_iter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L41-L67"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.calc_pval_onesided_ir" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.calc_pval_segmental_ir">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">calc_pval_segmental_ir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ori1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ori2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">src_score</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_iter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L173-L197"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.calc_pval_segmental_ir" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.get_best_holliday_junctions">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">get_best_holliday_junctions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pair</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">score_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[15,</span> <span class="pre">30,</span> <span class="pre">60]</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L276-L311"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.get_best_holliday_junctions" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.get_best_ir_within_breakpoints">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">get_best_ir_within_breakpoints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[15,</span> <span class="pre">30,</span> <span class="pre">60]</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L144-L170"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.get_best_ir_within_breakpoints" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.get_best_ir_within_segment">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">get_best_ir_within_segment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pair</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[15,</span> <span class="pre">30,</span> <span class="pre">60]</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L200-L230"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.get_best_ir_within_segment" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.get_best_onesided_ir">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">get_best_onesided_ir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">direction</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[15,</span> <span class="pre">30,</span> <span class="pre">60]</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L69-L103"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.get_best_onesided_ir" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.get_breakpoint_pair_seq_data">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">get_breakpoint_pair_seq_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pair</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L245-L252"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.get_breakpoint_pair_seq_data" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.get_onesided_ir">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">get_onesided_ir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sw</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L105-L117"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.get_onesided_ir" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.is_holliday_junction">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">is_holliday_junction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_cutoff2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_sum_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">score_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L233-L242"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.is_holliday_junction" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.irs.overlaps">
<span class="sig-prename descclassname"><span class="pre">ontmont.irs.</span></span><span class="sig-name descname"><span class="pre">overlaps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seg1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seg2</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/irs.py#L34-L39"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.irs.overlaps" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</section>
<section id="module-ontmont.plot">
<span id="ontmont-plot-module"></span><h2>ontmont.plot module<a class="headerlink" href="#module-ontmont.plot" title="Permalink to this heading">#</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.add_gene_annotations">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">add_gene_annotations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chromosome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">already_plotted</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L136-L171"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.add_gene_annotations" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.add_sv_legend_to_axes">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">add_sv_legend_to_axes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv_axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">svcolors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">supports</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0.82,</span> <span class="pre">0.5)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0.82,</span> <span class="pre">0.0)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alignment</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'left'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_support_legend</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L672-L696"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.add_sv_legend_to_axes" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.convert_tupleproxy_to_pyranges">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">convert_tupleproxy_to_pyranges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exon</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L38-L53"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.convert_tupleproxy_to_pyranges" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.extract_vaf_from_annotated_sv_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">extract_vaf_from_annotated_sv_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tra</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L615-L637"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.extract_vaf_from_annotated_sv_table" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.fix_coordinate_xticklabels">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">fix_coordinate_xticklabels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv_axes</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L539-L549"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.fix_coordinate_xticklabels" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.fix_yticks_with_integers">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">fix_yticks_with_integers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cn_axes</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L563-L572"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.fix_yticks_with_integers" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_chrom_xlims">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_chrom_xlims</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">read_svs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_chroms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_x_length_margin</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L407-L423"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_chrom_xlims" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_chromosomes_from_bundle">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_chromosomes_from_bundle</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L467-L474"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_chromosomes_from_bundle" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_cn_blocks_for_read_xlim">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_cn_blocks_for_read_xlim</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_chroms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_xlims</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clones</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['Pseudobulk']</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L385-L405"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_cn_blocks_for_read_xlim" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_edgecolor">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_edgecolor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">svtype</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L21-L29"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_edgecolor" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_gene_repr_exons">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_gene_repr_exons</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lenient</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L72-L75"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_gene_repr_exons" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_repr_transcript_id">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_repr_transcript_id</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lenient</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L84-L108"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_repr_transcript_id" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_sv_table_chroms">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_sv_table_chroms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L425-L428"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_sv_table_chroms" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_transcript_exons">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_transcript_exons</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L77-L82"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_transcript_exons" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.get_unique_sv_with_support">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">get_unique_sv_with_support</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bundle</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L726-L740"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.get_unique_sv_with_support" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.is_overlap">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">is_overlap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">interval1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interval2</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L31-L36"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.is_overlap" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.make_axes_for_breakpoints">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">make_axes_for_breakpoints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">brk_chroms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(10,</span> <span class="pre">4)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hspace</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rowsizes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(1,</span> <span class="pre">4,</span> <span class="pre">5,</span> <span class="pre">1)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">modes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['vaf',</span> <span class="pre">'cn',</span> <span class="pre">'sv',</span> <span class="pre">'gene']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width_ratios</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L476-L537"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.make_axes_for_breakpoints" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.make_spline_coordinates">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">make_spline_coordinates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pos1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_y</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin_div</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_offset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L698-L724"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.make_spline_coordinates" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.parse_gtf_region">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">parse_gtf_region</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L55-L70"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.parse_gtf_region" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_cn_on_axes">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_cn_on_axes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn_axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clone_colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metric</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'copy'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_cn</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L574-L613"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_cn_on_axes" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_gene_annotations">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_gene_annotations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_svs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L282-L300"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_gene_annotations" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_jabba_cn">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_jabba_cn</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_cn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn_colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ylim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0,</span> <span class="pre">6)</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L430-L444"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_jabba_cn" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_quasijabba">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_quasijabba</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_svs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn_clones</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['Pseudobulk']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(12,</span> <span class="pre">4)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">svlineh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width_ratios</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0.83,</span> <span class="pre">0.75)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">default_rad</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.25</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">suptitle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">linewidth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_support_legend</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_x_length_margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intra_y_offset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">linetick_divfactor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ylim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0,</span> <span class="pre">10)</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L233-L280"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_quasijabba" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_segments_on_axes">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_segments_on_axes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">plot_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_axes</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L781-L821"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_segments_on_axes" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_sv_lines_on_cn">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_sv_lines_on_cn</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">plot_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn_axes</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L551-L561"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_sv_lines_on_cn" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_svs_to_cn_segments">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_svs_to_cn_segments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_svs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_cns</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">svlineh</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">_linewidth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intra_y_offset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">linetick_divfactor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">default_rad</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.25</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_x_length_margin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_x_offset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100000</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L310-L383"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_svs_to_cn_segments" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_transition_on_axes">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_transition_on_axes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">plot_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L742-L779"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_transition_on_axes" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_vaf_cn_sv_for_reads">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_vaf_cn_sv_for_reads</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">plot_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clone_ids</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fig_row_sizes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(15,</span> <span class="pre">40,</span> <span class="pre">50)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn_window</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">200000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cn_metric</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'state'</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L450-L465"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_vaf_cn_sv_for_reads" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.plot.plot_vaf_on_axes">
<span class="sig-prename descclassname"><span class="pre">ontmont.plot.</span></span><span class="sig-name descname"><span class="pre">plot_vaf_on_axes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">plot_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vaf_axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clone_colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">flt_ont_sv</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/plot.py#L639-L670"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.plot.plot_vaf_on_axes" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</section>
<section id="module-ontmont.process">
<span id="ontmont-process-module"></span><h2>ontmont.process module<a class="headerlink" href="#module-ontmont.process" title="Permalink to this heading">#</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ontmont.process.filter_sv_by_clone">
<span class="sig-prename descclassname"><span class="pre">ontmont.process.</span></span><span class="sig-name descname"><span class="pre">filter_sv_by_clone</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vaf_col_str</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'vaf_'</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/process.py#L37-L48"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.process.filter_sv_by_clone" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.process.merge_cn_segments">
<span class="sig-prename descclassname"><span class="pre">ontmont.process.</span></span><span class="sig-name descname"><span class="pre">merge_cn_segments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">merge_gap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100000</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/process.py#L10-L35"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.process.merge_cn_segments" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.process.update_blocks_and_reset_prev">
<span class="sig-prename descclassname"><span class="pre">ontmont.process.</span></span><span class="sig-name descname"><span class="pre">update_blocks_and_reset_prev</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">blocks</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prev</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">features</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['clone_id',</span> <span class="pre">'chr',</span> <span class="pre">'start',</span> <span class="pre">'end',</span> <span class="pre">'state']</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/process.py#L3-L8"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.process.update_blocks_and_reset_prev" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</section>
<section id="module-ontmont.utils">
<span id="ontmont-utils-module"></span><h2>ontmont.utils module<a class="headerlink" href="#module-ontmont.utils" title="Permalink to this heading">#</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.enumerate_breakpoints">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">enumerate_breakpoints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L58-L83"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.enumerate_breakpoints" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.extract_split_alignments">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">extract_split_alignments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chroms_proc</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L157-L177"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.extract_split_alignments" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.filter_breakpoints_at_contig_ends">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">filter_breakpoints_at_contig_ends</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">brk_df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L86-L106"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.filter_breakpoints_at_contig_ends" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.filter_sv_with_breakpoint_at_contig_ends">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">filter_sv_with_breakpoint_at_contig_ends</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L109-L131"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.filter_sv_with_breakpoint_at_contig_ends" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.get_chromosomes_to_process">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">get_chromosomes_to_process</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_expression_vectors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L143-L154"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.get_chromosomes_to_process" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.get_secondaries">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">get_secondaries</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">read</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L134-L140"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.get_secondaries" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.is_breakpoints_not_sorted">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">is_breakpoints_not_sorted</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_order</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L198-L203"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.is_breakpoints_not_sorted" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.make_split_read_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">make_split_read_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alignments</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L179-L196"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.make_split_read_table" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.remove_duplicates_from_tra_table">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">remove_duplicates_from_tra_table</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tra_df</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L8-L55"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.remove_duplicates_from_tra_table" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.reverse_complement">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">reverse_complement</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L205-L208"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.reverse_complement" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ontmont.utils.shuffle_seq">
<span class="sig-prename descclassname"><span class="pre">ontmont.utils.</span></span><span class="sig-name descname"><span class="pre">shuffle_seq</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/shahcompbio/ontmont/blob/main/ontmont/utils.py#L210-L214"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ontmont.utils.shuffle_seq" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

</section>
<section id="module-ontmont">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-ontmont" title="Permalink to this heading">#</a></h2>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="index.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Welcome to ontmont’s documentation!</p>
      </div>
    </a>
    <a class="right-next"
       href="modules.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">ontmont</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
<div
    id="pst-page-navigation-heading-2"
    class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
  </div>
  <nav class="bd-toc-nav page-toc" aria-labelledby="pst-page-navigation-heading-2">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#submodules">Submodules</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.bundle">ontmont.bundle module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.bundle.make_brk_supports"><code class="docutils literal notranslate"><span class="pre">make_brk_supports()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.bundle.make_brk_table"><code class="docutils literal notranslate"><span class="pre">make_brk_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.bundle.make_brks_bundle"><code class="docutils literal notranslate"><span class="pre">make_brks_bundle()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.bundle.make_seg_table"><code class="docutils literal notranslate"><span class="pre">make_seg_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.bundle.make_tra_table"><code class="docutils literal notranslate"><span class="pre">make_tra_table()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.cli">ontmont.cli module</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.collect">ontmont.collect module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.extract_split_alignments"><code class="docutils literal notranslate"><span class="pre">extract_split_alignments()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.find_presence_of_matching_sv"><code class="docutils literal notranslate"><span class="pre">find_presence_of_matching_sv()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.fix_lower_support_coordinates"><code class="docutils literal notranslate"><span class="pre">fix_lower_support_coordinates()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.get_breakpoint_support_from_bundle"><code class="docutils literal notranslate"><span class="pre">get_breakpoint_support_from_bundle()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.get_normalized_sv"><code class="docutils literal notranslate"><span class="pre">get_normalized_sv()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.get_svtype"><code class="docutils literal notranslate"><span class="pre">get_svtype()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.make_brks_bundle"><code class="docutils literal notranslate"><span class="pre">make_brks_bundle()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.make_tumor_sv_table"><code class="docutils literal notranslate"><span class="pre">make_tumor_sv_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.map_similar_coordinate_to_higher_rank"><code class="docutils literal notranslate"><span class="pre">map_similar_coordinate_to_higher_rank()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.normalize_sv_table"><code class="docutils literal notranslate"><span class="pre">normalize_sv_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.pull_breakpoints_from_bam_files"><code class="docutils literal notranslate"><span class="pre">pull_breakpoints_from_bam_files()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.pull_breakpoints_from_reads_in_sv_regions"><code class="docutils literal notranslate"><span class="pre">pull_breakpoints_from_reads_in_sv_regions()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.collect.pull_sv_supporting_reads_from_bundle"><code class="docutils literal notranslate"><span class="pre">pull_sv_supporting_reads_from_bundle()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.datatypes">ontmont.datatypes module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Breakpoint"><code class="docutils literal notranslate"><span class="pre">Breakpoint</span></code></a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Breakpoint.chroms"><code class="docutils literal notranslate"><span class="pre">Breakpoint.chroms</span></code></a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Breakpoint.get_breakpoint_seqs"><code class="docutils literal notranslate"><span class="pre">Breakpoint.get_breakpoint_seqs()</span></code></a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.BreakpointChain"><code class="docutils literal notranslate"><span class="pre">BreakpointChain</span></code></a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.BreakpointChain.get_segments"><code class="docutils literal notranslate"><span class="pre">BreakpointChain.get_segments()</span></code></a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.BreakpointChain.get_transitions"><code class="docutils literal notranslate"><span class="pre">BreakpointChain.get_transitions()</span></code></a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.BreakpointPair"><code class="docutils literal notranslate"><span class="pre">BreakpointPair</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Sample"><code class="docutils literal notranslate"><span class="pre">Sample</span></code></a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Sample.long_samples"><code class="docutils literal notranslate"><span class="pre">Sample.long_samples</span></code></a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Sample.sample2group"><code class="docutils literal notranslate"><span class="pre">Sample.sample2group</span></code></a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Sample.sample2short"><code class="docutils literal notranslate"><span class="pre">Sample.sample2short</span></code></a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Segments"><code class="docutils literal notranslate"><span class="pre">Segments</span></code></a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Segments.get_list"><code class="docutils literal notranslate"><span class="pre">Segments.get_list()</span></code></a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.SplitAlignment"><code class="docutils literal notranslate"><span class="pre">SplitAlignment</span></code></a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.SplitAlignment.extract_cigar_field"><code class="docutils literal notranslate"><span class="pre">SplitAlignment.extract_cigar_field()</span></code></a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.SplitAlignment.get_cigar_tuples"><code class="docutils literal notranslate"><span class="pre">SplitAlignment.get_cigar_tuples()</span></code></a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Transitions"><code class="docutils literal notranslate"><span class="pre">Transitions</span></code></a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.Transitions.get_list"><code class="docutils literal notranslate"><span class="pre">Transitions.get_list()</span></code></a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.datatypes.get_breakpoint_seqs"><code class="docutils literal notranslate"><span class="pre">get_breakpoint_seqs()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.irs">ontmont.irs module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.align_two_sequences"><code class="docutils literal notranslate"><span class="pre">align_two_sequences()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.calc_distance_score"><code class="docutils literal notranslate"><span class="pre">calc_distance_score()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.calc_pval_bilateral_ir"><code class="docutils literal notranslate"><span class="pre">calc_pval_bilateral_ir()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.calc_pval_holliday"><code class="docutils literal notranslate"><span class="pre">calc_pval_holliday()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.calc_pval_onesided_ir"><code class="docutils literal notranslate"><span class="pre">calc_pval_onesided_ir()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.calc_pval_segmental_ir"><code class="docutils literal notranslate"><span class="pre">calc_pval_segmental_ir()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.get_best_holliday_junctions"><code class="docutils literal notranslate"><span class="pre">get_best_holliday_junctions()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.get_best_ir_within_breakpoints"><code class="docutils literal notranslate"><span class="pre">get_best_ir_within_breakpoints()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.get_best_ir_within_segment"><code class="docutils literal notranslate"><span class="pre">get_best_ir_within_segment()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.get_best_onesided_ir"><code class="docutils literal notranslate"><span class="pre">get_best_onesided_ir()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.get_breakpoint_pair_seq_data"><code class="docutils literal notranslate"><span class="pre">get_breakpoint_pair_seq_data()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.get_onesided_ir"><code class="docutils literal notranslate"><span class="pre">get_onesided_ir()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.is_holliday_junction"><code class="docutils literal notranslate"><span class="pre">is_holliday_junction()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.irs.overlaps"><code class="docutils literal notranslate"><span class="pre">overlaps()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.plot">ontmont.plot module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.add_gene_annotations"><code class="docutils literal notranslate"><span class="pre">add_gene_annotations()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.add_sv_legend_to_axes"><code class="docutils literal notranslate"><span class="pre">add_sv_legend_to_axes()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.convert_tupleproxy_to_pyranges"><code class="docutils literal notranslate"><span class="pre">convert_tupleproxy_to_pyranges()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.extract_vaf_from_annotated_sv_table"><code class="docutils literal notranslate"><span class="pre">extract_vaf_from_annotated_sv_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.fix_coordinate_xticklabels"><code class="docutils literal notranslate"><span class="pre">fix_coordinate_xticklabels()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.fix_yticks_with_integers"><code class="docutils literal notranslate"><span class="pre">fix_yticks_with_integers()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_chrom_xlims"><code class="docutils literal notranslate"><span class="pre">get_chrom_xlims()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_chromosomes_from_bundle"><code class="docutils literal notranslate"><span class="pre">get_chromosomes_from_bundle()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_cn_blocks_for_read_xlim"><code class="docutils literal notranslate"><span class="pre">get_cn_blocks_for_read_xlim()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_edgecolor"><code class="docutils literal notranslate"><span class="pre">get_edgecolor()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_gene_repr_exons"><code class="docutils literal notranslate"><span class="pre">get_gene_repr_exons()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_repr_transcript_id"><code class="docutils literal notranslate"><span class="pre">get_repr_transcript_id()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_sv_table_chroms"><code class="docutils literal notranslate"><span class="pre">get_sv_table_chroms()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_transcript_exons"><code class="docutils literal notranslate"><span class="pre">get_transcript_exons()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.get_unique_sv_with_support"><code class="docutils literal notranslate"><span class="pre">get_unique_sv_with_support()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.is_overlap"><code class="docutils literal notranslate"><span class="pre">is_overlap()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.make_axes_for_breakpoints"><code class="docutils literal notranslate"><span class="pre">make_axes_for_breakpoints()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.make_spline_coordinates"><code class="docutils literal notranslate"><span class="pre">make_spline_coordinates()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.parse_gtf_region"><code class="docutils literal notranslate"><span class="pre">parse_gtf_region()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_cn_on_axes"><code class="docutils literal notranslate"><span class="pre">plot_cn_on_axes()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_gene_annotations"><code class="docutils literal notranslate"><span class="pre">plot_gene_annotations()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_jabba_cn"><code class="docutils literal notranslate"><span class="pre">plot_jabba_cn()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_quasijabba"><code class="docutils literal notranslate"><span class="pre">plot_quasijabba()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_segments_on_axes"><code class="docutils literal notranslate"><span class="pre">plot_segments_on_axes()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_sv_lines_on_cn"><code class="docutils literal notranslate"><span class="pre">plot_sv_lines_on_cn()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_svs_to_cn_segments"><code class="docutils literal notranslate"><span class="pre">plot_svs_to_cn_segments()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_transition_on_axes"><code class="docutils literal notranslate"><span class="pre">plot_transition_on_axes()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_vaf_cn_sv_for_reads"><code class="docutils literal notranslate"><span class="pre">plot_vaf_cn_sv_for_reads()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.plot.plot_vaf_on_axes"><code class="docutils literal notranslate"><span class="pre">plot_vaf_on_axes()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.process">ontmont.process module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.process.filter_sv_by_clone"><code class="docutils literal notranslate"><span class="pre">filter_sv_by_clone()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.process.merge_cn_segments"><code class="docutils literal notranslate"><span class="pre">merge_cn_segments()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.process.update_blocks_and_reset_prev"><code class="docutils literal notranslate"><span class="pre">update_blocks_and_reset_prev()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont.utils">ontmont.utils module</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.enumerate_breakpoints"><code class="docutils literal notranslate"><span class="pre">enumerate_breakpoints()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.extract_split_alignments"><code class="docutils literal notranslate"><span class="pre">extract_split_alignments()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.filter_breakpoints_at_contig_ends"><code class="docutils literal notranslate"><span class="pre">filter_breakpoints_at_contig_ends()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.filter_sv_with_breakpoint_at_contig_ends"><code class="docutils literal notranslate"><span class="pre">filter_sv_with_breakpoint_at_contig_ends()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.get_chromosomes_to_process"><code class="docutils literal notranslate"><span class="pre">get_chromosomes_to_process()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.get_secondaries"><code class="docutils literal notranslate"><span class="pre">get_secondaries()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.is_breakpoints_not_sorted"><code class="docutils literal notranslate"><span class="pre">is_breakpoints_not_sorted()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.make_split_read_table"><code class="docutils literal notranslate"><span class="pre">make_split_read_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.remove_duplicates_from_tra_table"><code class="docutils literal notranslate"><span class="pre">remove_duplicates_from_tra_table()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.reverse_complement"><code class="docutils literal notranslate"><span class="pre">reverse_complement()</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ontmont.utils.shuffle_seq"><code class="docutils literal notranslate"><span class="pre">shuffle_seq()</span></code></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#module-ontmont">Module contents</a></li>
</ul>
  </nav></div>

  <div class="sidebar-secondary-item">

  <div class="tocsection sourcelink">
    <a href="_sources/ontmont.rst.txt">
      <i class="fa-solid fa-file-lines"></i> Show Source
    </a>
  </div>
</div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=3ee479438cf8b5e0d341"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=3ee479438cf8b5e0d341"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
      © Copyright 2024, Seongmin Choi.
      <br/>
    
  </p>
</div>
      
        <div class="footer-item">

  <p class="sphinx-version">
    Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 6.2.1.
    <br/>
  </p>
</div>
      
    </div>
  
  
  
    <div class="footer-items__end">
      
        <div class="footer-item">
<p class="theme-version">
  Built with the <a href="https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html">PyData Sphinx Theme</a> 0.15.3.
</p></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>