Search.setIndex({"docnames": ["index", "modules", "ontmont"], "filenames": ["index.rst", "modules.rst", "ontmont.rst"], "titles": ["Welcome to ontmont\u2019s documentation!", "ontmont", "ontmont package"], "terms": {"index": 0, "modul": [0, 1], "search": 0, "page": 0, "packag": 1, "submodul": 1, "bundl": 1, "cli": 1, "collect": 1, "datatyp": 1, "ir": 1, "plot": 1, "process": 1, "util": 1, "content": 1, "make_brk_support": 2, "make_brk_t": 2, "brk_support": 2, "make_brks_bundl": 2, "df": 2, "genom": 2, "sw_palindrom": 2, "sw_hollidai": 2, "margin": 2, "15": 2, "30": 2, "60": 2, "make_seg_t": 2, "seg_support": 2, "make_tra_t": 2, "tra_support": 2, "extract_split_align": 2, "read": 2, "max_depth": 2, "500": 2, "find_presence_of_matching_sv": 2, "sv1": 2, "sv2": 2, "50": 2, "fix_lower_support_coordin": 2, "complex": 2, "coord_map": 2, "get_breakpoint_support_from_bundl": 2, "get_normalized_sv": 2, "tra": 2, "get_svtyp": 2, "breakpointpair": 2, "make_tumor_sv_t": 2, "sv": 2, "10": 2, "get_support": 2, "true": 2, "map_similar_coordinate_to_higher_rank": 2, "breakpoint_support": 2, "normalize_sv_t": 2, "chrom1_col": 2, "chromosome_1": 2, "chrom2_col": 2, "chromosome_2": 2, "pos1_col": 2, "position_1": 2, "pos2_col": 2, "position_2": 2, "ori1_col": 2, "strand_1": 2, "ori2_col": 2, "strand_2": 2, "pull_breakpoints_from_bam_fil": 2, "bam_path": 2, "get_read_t": 2, "fals": 2, "pull_breakpoints_from_reads_in_sv_region": 2, "bam": 2, "min_n_breakpoint": 2, "3": 2, "pull_sv_supporting_reads_from_bundl": 2, "class": 2, "breakpoint": 2, "chrom": 2, "po": 2, "orient": 2, "base": 2, "object": 2, "1": 2, "2": 2, "4": 2, "5": 2, "6": 2, "7": 2, "8": 2, "9": 2, "11": 2, "12": 2, "13": 2, "14": 2, "16": 2, "17": 2, "18": 2, "19": 2, "20": 2, "21": 2, "22": 2, "x": 2, "y": 2, "m": 2, "chr1": 2, "chr2": 2, "chr3": 2, "chr4": 2, "chr5": 2, "chr6": 2, "chr7": 2, "chr8": 2, "chr9": 2, "chr10": 2, "chr11": 2, "chr12": 2, "chr13": 2, "chr14": 2, "chr15": 2, "chr16": 2, "chr17": 2, "chr18": 2, "chr19": 2, "chr20": 2, "chr21": 2, "chr22": 2, "chrx": 2, "chry": 2, "chrm": 2, "get_breakpoint_seq": 2, "breakpointchain": 2, "brks_iter": 2, "list": 2, "get_seg": 2, "get_transit": 2, "sort_transit": 2, "brk1": 2, "brk2": 2, "sampl": 2, "long_sampl": 2, "14472b_201": 2, "14472b_202": 2, "14472b_500": 2, "14472b_501": 2, "14472c_203": 2, "14472c_204": 2, "14472c_502": 2, "14472d_101": 2, "14472d_102": 2, "14472d_103": 2, "14472d_104": 2, "14472d_205": 2, "14472_100": 2, "14472_300": 2, "sample2group": 2, "sample2short": 2, "201": 2, "202": 2, "501": 2, "203": 2, "204": 2, "502": 2, "101": 2, "102": 2, "103": 2, "104": 2, "205": 2, "100": 2, "300": 2, "segment": 2, "get_list": 2, "splitalign": 2, "cigarstr": 2, "read_nam": 2, "refnam": 2, "read_po": 2, "strand": 2, "repres": 2, "split": 2, "align": 2, "sequenc": 2, "encapsul": 2, "detail": 2, "about": 2, "": 2, "locat": 2, "structur": 2, "describ": 2, "its": 2, "cigar": 2, "string": 2, "thi": 2, "i": 2, "design": 2, "pars": 2, "store": 2, "inform": 2, "singl": 2, "which": 2, "can": 2, "part": 2, "primari": 2, "secondari": 2, "chimer": 2, "It": 2, "includ": 2, "method": 2, "interpret": 2, "further": 2, "analysi": 2, "properti": 2, "calcul": 2, "length": 2, "identifi": 2, "specif": 2, "featur": 2, "extract_cigar_field": 2, "static": 2, "get_cigar_tupl": 2, "return": 2, "tupl": 2, "from": 2, "transit": 2, "align_two_sequ": 2, "seq1": 2, "seq2": 2, "sw": 2, "rc": 2, "given": 2, "revers": 2, "complement": 2, "default": 2, "calc_distance_scor": 2, "dist1": 2, "dist2": 2, "dist_cutoff1": 2, "dist_cutoff2": 2, "calc_pval_bilateral_ir": 2, "src_score": 2, "n_iter": 2, "1000": 2, "random_se": 2, "42": 2, "calc_pval_hollidai": 2, "negative1": 2, "negative2": 2, "calc_pval_onesided_ir": 2, "seq": 2, "direct": 2, "dist_cutoff": 2, "calc_pval_segmental_ir": 2, "ori1": 2, "ori2": 2, "get_best_holliday_junct": 2, "pair": 2, "score_cutoff": 2, "get_best_ir_within_breakpoint": 2, "get_best_ir_within_seg": 2, "get_best_onesided_ir": 2, "get_breakpoint_pair_seq_data": 2, "get_onesided_ir": 2, "one": 2, "seq_rc": 2, "find": 2, "invert": 2, "repeat": 2, "is_holliday_junct": 2, "aln": 2, "dist_sum_cutoff": 2, "overlap": 2, "seg1": 2, "seg2": 2, "add_gene_annot": 2, "ax": 2, "gtf": 2, "chromosom": 2, "start": 2, "end": 2, "gene": 2, "none": 2, "already_plot": 2, "gene_font_s": 2, "ucsc": 2, "like": 2, "annot": 2, "pyplot": 2, "pysam": 2, "tabixfil": 2, "input": 2, "after": 2, "open": 2, "tabix": 2, "gz": 2, "str": 2, "int": 2, "option": 2, "return_offset": 2, "offset": 2, "map": 2, "dict": 2, "gene2offset": 2, "add_sv_legend_to_ax": 2, "sv_ax": 2, "svcolor": 2, "support": 2, "alpha": 2, "0": 2, "loc1": 2, "82": 2, "loc2": 2, "size": 2, "left": 2, "show_support_legend": 2, "convert_tupleproxy_to_pyrang": 2, "exon": 2, "convert": 2, "tupleproxi": 2, "pyrang": 2, "field": 2, "result": 2, "iter": 2, "extract_vaf_from_annotated_sv_t": 2, "fix_coordinate_xticklabel": 2, "fix_yticks_with_integ": 2, "cn_ax": 2, "get_chrom_xlim": 2, "read_sv": 2, "read_chrom": 2, "sv_x_length_margin": 2, "get_chromosomes_from_bundl": 2, "get_cn_blocks_for_read_xlim": 2, "cn": 2, "chrom_xlim": 2, "bin_siz": 2, "50000": 2, "clone": 2, "pseudobulk": 2, "get_edgecolor": 2, "svtype": 2, "get_gene_repr_exon": 2, "lenient": 2, "get": 2, "name": 2, "datafram": 2, "symbol": 2, "isoform": 2, "give": 2, "get_repr_transcript_id": 2, "gene_nam": 2, "select": 2, "transcript": 2, "transcript_id": 2, "get_sv_table_chrom": 2, "get_transcript_exon": 2, "get_unique_sv_with_support": 2, "is_overlap": 2, "interval1": 2, "interval2": 2, "statu": 2, "two": 2, "interv": 2, "test": 2, "follow": 2, "otherwis": 2, "make_axes_for_breakpoint": 2, "brk_chrom": 2, "figsiz": 2, "hspace": 2, "rowsiz": 2, "mode": 2, "vaf": 2, "width_ratio": 2, "make_spline_coordin": 2, "pos1": 2, "pos2": 2, "sv_y": 2, "margin_div": 2, "y_offset": 2, "05": 2, "parse_gtf_region": 2, "region": 2, "fetch": 2, "quasi": 2, "pd": 2, "plot_cn_on_ax": 2, "clone_color": 2, "metric": 2, "copi": 2, "max_cn": 2, "plot_gene_annot": 2, "input_sv": 2, "plot_jabba_cn": 2, "chrom_cn": 2, "cn_color": 2, "ylim": 2, "plot_quasijabba": 2, "cn_clone": 2, "svlineh": 2, "83": 2, "75": 2, "default_rad": 2, "25": 2, "suptitl": 2, "read_length": 2, "gene_margin": 2, "linewidth": 2, "intra_y_offset": 2, "linetick_divfactor": 2, "plot_segments_on_ax": 2, "plot_data": 2, "plot_sv_lines_on_cn": 2, "plot_svs_to_cn_seg": 2, "_linewidth": 2, "sv_x_offset": 2, "100000": 2, "plot_transition_on_ax": 2, "plot_vaf_cn_sv_for_read": 2, "clone_id": 2, "fig_row_s": 2, "40": 2, "cn_window": 2, "200000": 2, "cn_metric": 2, "state": 2, "plot_vaf_on_ax": 2, "vaf_ax": 2, "flt_ont_sv": 2, "filter_sv_by_clon": 2, "vaf_col_str": 2, "vaf_": 2, "merge_cn_seg": 2, "merge_gap": 2, "update_blocks_and_reset_prev": 2, "block": 2, "prev": 2, "row": 2, "chr": 2, "enumerate_breakpoint": 2, "chroms_proc": 2, "extract": 2, "organ": 2, "file": 2, "specifi": 2, "function": 2, "over": 2, "each": 2, "focus": 2, "have": 2, "indic": 2, "construct": 2, "contain": 2, "queri": 2, "refer": 2, "posit": 2, "paramet": 2, "alignmentfil": 2, "an": 2, "A": 2, "note": 2, "assum": 2, "exist": 2, "get_secondari": 2, "without": 2, "ar": 2, "skip": 2, "filter_breakpoints_at_contig_end": 2, "brk_df": 2, "filter_sv_with_breakpoint_at_contig_end": 2, "filter": 2, "variant": 2, "record": 2, "exclud": 2, "those": 2, "too": 2, "close": 2, "contig": 2, "distanc": 2, "nearest": 2, "respect": 2, "either": 2, "within": 2, "The": 2, "us": 2, "predefin": 2, "dictionari": 2, "determin": 2, "panda": 2, "must": 2, "chrom1": 2, "column": 2, "where": 2, "get_chromosomes_to_process": 2, "drop_expression_vector": 2, "least": 2, "both": 2, "present": 2, "vector": 2, "check": 2, "presenc": 2, "befor": 2, "ad": 2, "tag": 2, "sa": 2, "expect": 2, "semicolon": 2, "separ": 2, "These": 2, "element": 2, "alignedseg": 2, "sam": 2, "provid": 2, "comma": 2, "format": 2, "empti": 2, "found": 2, "is_breakpoints_not_sort": 2, "chrom2": 2, "chrom_ord": 2, "whether": 2, "order": 2, "correctli": 2, "make_split_read_t": 2, "creat": 2, "take": 2, "transform": 2, "clip": 2, "match": 2, "sort": 2, "first": 2, "duplic": 2, "remov": 2, "ensur": 2, "uniqu": 2, "entri": 2, "attribut": 2, "clip1": 2, "clip2": 2, "pclip1": 2, "qname": 2, "remove_duplicates_from_tra_t": 2, "tra_df": 2, "reverse_compl": 2, "shuffle_seq": 2}, "objects": {"": [[2, 0, 0, "-", "ontmont"]], "ontmont": [[2, 0, 0, "-", "bundle"], [2, 0, 0, "-", "cli"], [2, 0, 0, "-", "collect"], [2, 0, 0, "-", "datatypes"], [2, 0, 0, "-", "irs"], [2, 0, 0, "-", "plot"], [2, 0, 0, "-", "process"], [2, 0, 0, "-", "utils"]], "ontmont.bundle": [[2, 1, 1, "", "make_brk_supports"], [2, 1, 1, "", "make_brk_table"], [2, 1, 1, "", "make_brks_bundle"], [2, 1, 1, "", "make_seg_table"], [2, 1, 1, "", "make_tra_table"]], "ontmont.collect": [[2, 1, 1, "", "extract_split_alignments"], [2, 1, 1, "", "find_presence_of_matching_sv"], [2, 1, 1, "", "fix_lower_support_coordinates"], [2, 1, 1, "", "get_breakpoint_support_from_bundle"], [2, 1, 1, "", "get_normalized_sv"], [2, 1, 1, "", "get_svtype"], [2, 1, 1, "", "make_brks_bundle"], [2, 1, 1, "", "make_tumor_sv_table"], [2, 1, 1, "", "map_similar_coordinate_to_higher_rank"], [2, 1, 1, "", "normalize_sv_table"], [2, 1, 1, "", "pull_breakpoints_from_bam_files"], [2, 1, 1, "", "pull_breakpoints_from_reads_in_sv_regions"], [2, 1, 1, "", "pull_sv_supporting_reads_from_bundle"]], "ontmont.datatypes": [[2, 2, 1, "", "Breakpoint"], [2, 2, 1, "", "BreakpointChain"], [2, 2, 1, "", "BreakpointPair"], [2, 2, 1, "", "Sample"], [2, 2, 1, "", "Segments"], [2, 2, 1, "", "SplitAlignment"], [2, 2, 1, "", "Transitions"], [2, 1, 1, "", "get_breakpoint_seqs"]], "ontmont.datatypes.Breakpoint": [[2, 3, 1, "", "chroms"], [2, 4, 1, "", "get_breakpoint_seqs"]], "ontmont.datatypes.BreakpointChain": [[2, 4, 1, "", "get_segments"], [2, 4, 1, "", "get_transitions"]], "ontmont.datatypes.Sample": [[2, 3, 1, "", "long_samples"], [2, 3, 1, "", "sample2group"], [2, 3, 1, "", "sample2short"]], "ontmont.datatypes.Segments": [[2, 4, 1, "", "get_list"]], "ontmont.datatypes.SplitAlignment": [[2, 4, 1, "", "extract_cigar_field"], [2, 4, 1, "", "get_cigar_tuples"]], "ontmont.datatypes.Transitions": [[2, 4, 1, "", "get_list"]], "ontmont.irs": [[2, 1, 1, "", "align_two_sequences"], [2, 1, 1, "", "calc_distance_score"], [2, 1, 1, "", "calc_pval_bilateral_ir"], [2, 1, 1, "", "calc_pval_holliday"], [2, 1, 1, "", "calc_pval_onesided_ir"], [2, 1, 1, "", "calc_pval_segmental_ir"], [2, 1, 1, "", "get_best_holliday_junctions"], [2, 1, 1, "", "get_best_ir_within_breakpoints"], [2, 1, 1, "", "get_best_ir_within_segment"], [2, 1, 1, "", "get_best_onesided_ir"], [2, 1, 1, "", "get_breakpoint_pair_seq_data"], [2, 1, 1, "", "get_onesided_ir"], [2, 1, 1, "", "is_holliday_junction"], [2, 1, 1, "", "overlaps"]], "ontmont.plot": [[2, 1, 1, "", "add_gene_annotations"], [2, 1, 1, "", "add_sv_legend_to_axes"], [2, 1, 1, "", "convert_tupleproxy_to_pyranges"], [2, 1, 1, "", "extract_vaf_from_annotated_sv_table"], [2, 1, 1, "", "fix_coordinate_xticklabels"], [2, 1, 1, "", "fix_yticks_with_integers"], [2, 1, 1, "", "get_chrom_xlims"], [2, 1, 1, "", "get_chromosomes_from_bundle"], [2, 1, 1, "", "get_cn_blocks_for_read_xlim"], [2, 1, 1, "", "get_edgecolor"], [2, 1, 1, "", "get_gene_repr_exons"], [2, 1, 1, "", "get_repr_transcript_id"], [2, 1, 1, "", "get_sv_table_chroms"], [2, 1, 1, "", "get_transcript_exons"], [2, 1, 1, "", "get_unique_sv_with_support"], [2, 1, 1, "", "is_overlap"], [2, 1, 1, "", "make_axes_for_breakpoints"], [2, 1, 1, "", "make_spline_coordinates"], [2, 1, 1, "", "parse_gtf_region"], [2, 1, 1, "", "plot_cn_on_axes"], [2, 1, 1, "", "plot_gene_annotations"], [2, 1, 1, "", "plot_jabba_cn"], [2, 1, 1, "", "plot_quasijabba"], [2, 1, 1, "", "plot_segments_on_axes"], [2, 1, 1, "", "plot_sv_lines_on_cn"], [2, 1, 1, "", "plot_svs_to_cn_segments"], [2, 1, 1, "", "plot_transition_on_axes"], [2, 1, 1, "", "plot_vaf_cn_sv_for_reads"], [2, 1, 1, "", "plot_vaf_on_axes"]], "ontmont.process": [[2, 1, 1, "", "filter_sv_by_clone"], [2, 1, 1, "", "merge_cn_segments"], [2, 1, 1, "", "update_blocks_and_reset_prev"]], "ontmont.utils": [[2, 1, 1, "", "enumerate_breakpoints"], [2, 1, 1, "", "extract_split_alignments"], [2, 1, 1, "", "filter_breakpoints_at_contig_ends"], [2, 1, 1, "", "filter_sv_with_breakpoint_at_contig_ends"], [2, 1, 1, "", "get_chromosomes_to_process"], [2, 1, 1, "", "get_secondaries"], [2, 1, 1, "", "is_breakpoints_not_sorted"], [2, 1, 1, "", "make_split_read_table"], [2, 1, 1, "", "remove_duplicates_from_tra_table"], [2, 1, 1, "", "reverse_complement"], [2, 1, 1, "", "shuffle_seq"]]}, "objtypes": {"0": "py:module", "1": "py:function", "2": "py:class", "3": "py:attribute", "4": "py:method"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"], "2": ["py", "class", "Python class"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "method", "Python method"]}, "titleterms": {"welcom": 0, "ontmont": [0, 1, 2], "": 0, "document": 0, "indic": 0, "tabl": 0, "packag": 2, "submodul": 2, "bundl": 2, "modul": 2, "cli": 2, "collect": 2, "datatyp": 2, "ir": 2, "plot": 2, "process": 2, "util": 2, "content": 2}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 6, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 56}})