Search.setIndex({"docnames": ["index", "modules", "ontmont"], "filenames": ["index.rst", "modules.rst", "ontmont.rst"], "titles": ["Welcome to ontmont\u2019s documentation!", "ontmont", "ontmont package"], "terms": {"packag": [0, 1], "submodul": [0, 1], "bundl": [0, 1], "modul": [0, 1], "cli": [0, 1], "collect": [0, 1], "datatyp": [0, 1], "ir": [0, 1], "plot": [0, 1], "process": [0, 1], "util": [0, 1], "index": 0, "search": 0, "page": 0, "make_brk_support": [1, 2], "make_brk_t": [1, 2], "make_brks_bundl": [1, 2], "make_seg_t": [1, 2], "make_tra_t": [1, 2], "extract_split_align": [1, 2], "find_presence_of_matching_sv": [1, 2], "fix_lower_support_coordin": [1, 2], "get_breakpoint_support_from_bundl": [1, 2], "get_normalized_sv": [1, 2], "get_svtyp": [1, 2], "make_tumor_sv_t": [1, 2], "map_similar_coordinate_to_higher_rank": [1, 2], "normalize_sv_t": [1, 2], "pull_breakpoints_from_bam_fil": [1, 2], "pull_breakpoints_from_reads_in_sv_region": [1, 2], "pull_sv_supporting_reads_from_bundl": [1, 2], "breakpoint": [1, 2], "chrom": [1, 2], "get_breakpoint_seq": [1, 2], "breakpointchain": [1, 2], "get_seg": [1, 2], "get_transit": [1, 2], "breakpointpair": [1, 2], "sampl": [1, 2], "long_sampl": [1, 2], "sample2group": [1, 2], "sample2short": [1, 2], "segment": [1, 2], "get_list": [1, 2], "splitalign": [1, 2], "extract_cigar_field": [1, 2], "get_cigar_tupl": [1, 2], "transit": [1, 2], "align_two_sequ": [1, 2], "calc_distance_scor": [1, 2], "calc_pval_bilateral_ir": [1, 2], "calc_pval_hollidai": [1, 2], "calc_pval_onesided_ir": [1, 2], "calc_pval_segmental_ir": [1, 2], "get_best_holliday_junct": [1, 2], "get_best_ir_within_breakpoint": [1, 2], "get_best_ir_within_seg": [1, 2], "get_best_onesided_ir": [1, 2], "get_breakpoint_pair_seq_data": [1, 2], "get_onesided_ir": [1, 2], "is_holliday_junct": [1, 2], "overlap": [1, 2], "add_gene_annot": [1, 2], "add_sv_legend_to_ax": [1, 2], "convert_tupleproxy_to_pyrang": [1, 2], "extract_vaf_from_annotated_sv_t": [1, 2], "fix_coordinate_xticklabel": [1, 2], "fix_yticks_with_integ": [1, 2], "get_chrom_xlim": [1, 2], "get_chromosomes_from_bundl": [1, 2], "get_cn_blocks_for_read_xlim": [1, 2], "get_edgecolor": [1, 2], "get_gene_repr_exon": [1, 2], "get_repr_transcript_id": [1, 2], "get_sv_table_chrom": [1, 2], "get_transcript_exon": [1, 2], "get_unique_sv_with_support": [1, 2], "is_overlap": [1, 2], "make_axes_for_breakpoint": [1, 2], "make_spline_coordin": [1, 2], "parse_gtf_region": [1, 2], "plot_cn_on_ax": [1, 2], "plot_gene_annot": [1, 2], "plot_jabba_cn": [1, 2], "plot_quasijabba": [1, 2], "plot_segments_on_ax": [1, 2], "plot_sv_lines_on_cn": [1, 2], "plot_svs_to_cn_seg": [1, 2], "plot_transition_on_ax": [1, 2], "plot_vaf_cn_sv_for_read": [1, 2], "plot_vaf_on_ax": [1, 2], "filter_sv_by_clon": [1, 2], "merge_cn_seg": [1, 2], "update_blocks_and_reset_prev": [1, 2], "enumerate_breakpoint": [1, 2], "filter_breakpoints_at_contig_end": [1, 2], "filter_sv_with_breakpoint_at_contig_end": [1, 2], "get_chromosomes_to_process": [1, 2], "get_secondari": [1, 2], "is_breakpoints_not_sort": [1, 2], "make_split_read_t": [1, 2], "remove_duplicates_from_tra_t": [1, 2], "reverse_compl": [1, 2], "shuffle_seq": [1, 2], "content": 1, "sourc": 2, "count": 2, "support": 2, "uniqu": 2, "coordin": 2, "paramet": 2, "list": 2, "return": 2, "panda": 2, "seri": 2, "take": 2, "po": 2, "ori": 2, "account": 2, "brk_support": 2, "unilateral_score_cutoff": 2, "5": 2, "bilateral_score_cutoff": 2, "8": 2, "creat": 2, "datafram": 2, "variabl": 2, "dict": 2, "number": 2, "each": 2, "int": 2, "option": 2, "align": 2, "score": 2, "cutoff": 2, "unilater": 2, "default": 2, "bilater": 2, "tabl": 2, "statist": 2, "reads_df": 2, "genom": 2, "sw_palindrom": 2, "sw_hollidai": 2, "margin": 2, "15": 2, "30": 2, "60": 2, "make": 2, "breapointchain": 2, "base": 2, "read": 2, "pyfaidx": 2, "fasta": 2, "swalign": 2, "localalign": 2, "detect": 2, "homologi": 2, "slice": 2, "from": 2, "seg_support": 2, "segment_score_cutoff": 2, "found": 2, "tra_support": 2, "sv": 2, "duplic": 2, "remov": 2, "locat": 2, "contig": 2, "termini": 2, "max_read": 2, "500": 2, "extract": 2, "object": 2, "iteratorrow": 2, "pysam": 2, "fetch": 2, "alignmentfil": 2, "maximum": 2, "sv1": 2, "sv2": 2, "50": 2, "check": 2, "label": 2, "match": 2, "refer": 2, "presenc": 2, "bp": 2, "differ": 2, "pd": 2, "true": 2, "fals": 2, "length": 2, "equal": 2, "row": 2, "size": 2, "complex": 2, "coord_map": 2, "map": 2, "lower": 2, "close": 2, "higher": 2, "str": 2, "fix": 2, "get": 2, "counter": 2, "tra": 2, "sort": 2, "normal": 2, "pair": 2, "flatten": 2, "type": 2, "string": 2, "given": 2, "rais": 2, "valueerror": 2, "If": 2, "ha": 2, "been": 2, "assign": 2, "none": 2, "10": 2, "get_support": 2, "in_sourc": 2, "flag": 2, "merg": 2, "cluster": 2, "bool": 2, "same": 2, "add": 2, "collaps": 2, "breakpoint_support": 2, "prioriti": 2, "tupl": 2, "contain": 2, "src": 2, "dst": 2, "coord_map_log": 2, "max_coord": 2, "src_count": 2, "max_count": 2, "onli": 2, "debug": 2, "chrom1_col": 2, "chromosome_1": 2, "chrom2_col": 2, "chromosome_2": 2, "pos1_col": 2, "position_1": 2, "pos2_col": 2, "position_2": 2, "ori1_col": 2, "strand_1": 2, "ori2_col": 2, "strand_2": 2, "breakpoint1": 2, "breakpoint2": 2, "bam_path": 2, "get_read_t": 2, "bam": 2, "file": 2, "accord": 2, "an": 2, "input": 2, "well": 2, "stat": 2, "min_n_breakpoint": 2, "3": 2, "append": 2, "minimum": 2, "requir": 2, "save": 2, "us": 2, "select": 2, "rearrang": 2, "i": 2, "high": 2, "filter": 2, "includ": 2, "have": 2, "class": 2, "orient": 2, "1": 2, "2": 2, "4": 2, "6": 2, "7": 2, "9": 2, "11": 2, "12": 2, "13": 2, "14": 2, "16": 2, "17": 2, "18": 2, "19": 2, "20": 2, "21": 2, "22": 2, "x": 2, "y": 2, "m": 2, "chr1": 2, "chr2": 2, "chr3": 2, "chr4": 2, "chr5": 2, "chr6": 2, "chr7": 2, "chr8": 2, "chr9": 2, "chr10": 2, "chr11": 2, "chr12": 2, "chr13": 2, "chr14": 2, "chr15": 2, "chr16": 2, "chr17": 2, "chr18": 2, "chr19": 2, "chr20": 2, "chr21": 2, "chr22": 2, "chrx": 2, "chry": 2, "chrm": 2, "brks_iter": 2, "sort_transit": 2, "brk1": 2, "brk2": 2, "14472b_201": 2, "14472b_202": 2, "14472b_500": 2, "14472b_501": 2, "14472c_203": 2, "14472c_204": 2, "14472c_502": 2, "14472d_101": 2, "14472d_102": 2, "14472d_103": 2, "14472d_104": 2, "14472d_205": 2, "14472_100": 2, "14472_300": 2, "201": 2, "202": 2, "501": 2, "203": 2, "204": 2, "502": 2, "101": 2, "102": 2, "103": 2, "104": 2, "205": 2, "100": 2, "300": 2, "df": 2, "cigarstr": 2, "read_nam": 2, "refnam": 2, "read_po": 2, "strand": 2, "static": 2, "cigar": 2, "seq1": 2, "seq2": 2, "sw": 2, "rc": 2, "dist1": 2, "dist2": 2, "dist_cutoff1": 2, "dist_cutoff2": 2, "src_score": 2, "n_iter": 2, "1000": 2, "random_se": 2, "42": 2, "negative1": 2, "negative2": 2, "seq": 2, "direct": 2, "dist_cutoff": 2, "ori1": 2, "ori2": 2, "score_cutoff": 2, "aln": 2, "dist_sum_cutoff": 2, "seg1": 2, "seg2": 2, "ax": 2, "gtf": 2, "chromosom": 2, "start": 2, "end": 2, "gene": 2, "already_plot": 2, "gene_font_s": 2, "sv_ax": 2, "svcolor": 2, "alpha": 2, "0": 2, "loc1": 2, "82": 2, "loc2": 2, "left": 2, "show_support_legend": 2, "exon": 2, "cn_ax": 2, "read_sv": 2, "read_chrom": 2, "sv_x_length_margin": 2, "cn": 2, "chrom_xlim": 2, "bin_siz": 2, "50000": 2, "clone": 2, "pseudobulk": 2, "svtype": 2, "lenient": 2, "gene_nam": 2, "transcript_id": 2, "interval1": 2, "interval2": 2, "brk_chrom": 2, "figsiz": 2, "hspace": 2, "rowsiz": 2, "mode": 2, "vaf": 2, "width_ratio": 2, "pos1": 2, "pos2": 2, "sv_y": 2, "margin_div": 2, "y_offset": 2, "05": 2, "region": 2, "clone_color": 2, "metric": 2, "copi": 2, "max_cn": 2, "input_sv": 2, "chrom_cn": 2, "cn_color": 2, "ylim": 2, "cn_clone": 2, "svlineh": 2, "83": 2, "75": 2, "default_rad": 2, "25": 2, "suptitl": 2, "read_length": 2, "gene_margin": 2, "linewidth": 2, "intra_y_offset": 2, "linetick_divfactor": 2, "plot_data": 2, "_linewidth": 2, "sv_x_offset": 2, "100000": 2, "clone_id": 2, "fig_row_s": 2, "40": 2, "cn_window": 2, "200000": 2, "cn_metric": 2, "state": 2, "vaf_ax": 2, "flt_ont_sv": 2, "vaf_col_str": 2, "vaf_": 2, "merge_gap": 2, "block": 2, "prev": 2, "featur": 2, "chr": 2, "chroms_proc": 2, "brk_df": 2, "drop_expression_vector": 2, "chrom1": 2, "chrom2": 2, "chrom_ord": 2, "tra_df": 2}, "objects": {"": [[2, 0, 0, "-", "ontmont"]], "ontmont": [[2, 0, 0, "-", "bundle"], [2, 0, 0, "-", "cli"], [2, 0, 0, "-", "collect"], [2, 0, 0, "-", "datatypes"], [2, 0, 0, "-", "irs"], [2, 0, 0, "-", "plot"], [2, 0, 0, "-", "process"], [2, 0, 0, "-", "utils"]], "ontmont.bundle": [[2, 1, 1, "", "make_brk_supports"], [2, 1, 1, "", "make_brk_table"], [2, 1, 1, "", "make_brks_bundle"], [2, 1, 1, "", "make_seg_table"], [2, 1, 1, "", "make_tra_table"]], "ontmont.collect": [[2, 1, 1, "", "extract_split_alignments"], [2, 1, 1, "", "find_presence_of_matching_sv"], [2, 1, 1, "", "fix_lower_support_coordinates"], [2, 1, 1, "", "get_breakpoint_support_from_bundle"], [2, 1, 1, "", "get_normalized_sv"], [2, 1, 1, "", "get_svtype"], [2, 1, 1, "", "make_brks_bundle"], [2, 1, 1, "", "make_tumor_sv_table"], [2, 1, 1, "", "map_similar_coordinate_to_higher_rank"], [2, 1, 1, "", "normalize_sv_table"], [2, 1, 1, "", "pull_breakpoints_from_bam_files"], [2, 1, 1, "", "pull_breakpoints_from_reads_in_sv_regions"], [2, 1, 1, "", "pull_sv_supporting_reads_from_bundle"]], "ontmont.datatypes": [[2, 2, 1, "", "Breakpoint"], [2, 2, 1, "", "BreakpointChain"], [2, 2, 1, "", "BreakpointPair"], [2, 2, 1, "", "Sample"], [2, 2, 1, "", "Segments"], [2, 2, 1, "", "SplitAlignment"], [2, 2, 1, "", "Transitions"], [2, 1, 1, "", "get_breakpoint_seqs"]], "ontmont.datatypes.Breakpoint": [[2, 3, 1, "", "chroms"], [2, 4, 1, "", "get_breakpoint_seqs"]], "ontmont.datatypes.BreakpointChain": [[2, 4, 1, "", "get_segments"], [2, 4, 1, "", "get_transitions"]], "ontmont.datatypes.Sample": [[2, 3, 1, "", "long_samples"], [2, 3, 1, "", "sample2group"], [2, 3, 1, "", "sample2short"]], "ontmont.datatypes.Segments": [[2, 4, 1, "", "get_list"]], "ontmont.datatypes.SplitAlignment": [[2, 4, 1, "", "extract_cigar_field"], [2, 4, 1, "", "get_cigar_tuples"]], "ontmont.datatypes.Transitions": [[2, 4, 1, "", "get_list"]], "ontmont.irs": [[2, 1, 1, "", "align_two_sequences"], [2, 1, 1, "", "calc_distance_score"], [2, 1, 1, "", "calc_pval_bilateral_ir"], [2, 1, 1, "", "calc_pval_holliday"], [2, 1, 1, "", "calc_pval_onesided_ir"], [2, 1, 1, "", "calc_pval_segmental_ir"], [2, 1, 1, "", "get_best_holliday_junctions"], [2, 1, 1, "", "get_best_ir_within_breakpoints"], [2, 1, 1, "", "get_best_ir_within_segment"], [2, 1, 1, "", "get_best_onesided_ir"], [2, 1, 1, "", "get_breakpoint_pair_seq_data"], [2, 1, 1, "", "get_onesided_ir"], [2, 1, 1, "", "is_holliday_junction"], [2, 1, 1, "", "overlaps"]], "ontmont.plot": [[2, 1, 1, "", "add_gene_annotations"], [2, 1, 1, "", "add_sv_legend_to_axes"], [2, 1, 1, "", "convert_tupleproxy_to_pyranges"], [2, 1, 1, "", "extract_vaf_from_annotated_sv_table"], [2, 1, 1, "", "fix_coordinate_xticklabels"], [2, 1, 1, "", "fix_yticks_with_integers"], [2, 1, 1, "", "get_chrom_xlims"], [2, 1, 1, "", "get_chromosomes_from_bundle"], [2, 1, 1, "", "get_cn_blocks_for_read_xlim"], [2, 1, 1, "", "get_edgecolor"], [2, 1, 1, "", "get_gene_repr_exons"], [2, 1, 1, "", "get_repr_transcript_id"], [2, 1, 1, "", "get_sv_table_chroms"], [2, 1, 1, "", "get_transcript_exons"], [2, 1, 1, "", "get_unique_sv_with_support"], [2, 1, 1, "", "is_overlap"], [2, 1, 1, "", "make_axes_for_breakpoints"], [2, 1, 1, "", "make_spline_coordinates"], [2, 1, 1, "", "parse_gtf_region"], [2, 1, 1, "", "plot_cn_on_axes"], [2, 1, 1, "", "plot_gene_annotations"], [2, 1, 1, "", "plot_jabba_cn"], [2, 1, 1, "", "plot_quasijabba"], [2, 1, 1, "", "plot_segments_on_axes"], [2, 1, 1, "", "plot_sv_lines_on_cn"], [2, 1, 1, "", "plot_svs_to_cn_segments"], [2, 1, 1, "", "plot_transition_on_axes"], [2, 1, 1, "", "plot_vaf_cn_sv_for_reads"], [2, 1, 1, "", "plot_vaf_on_axes"]], "ontmont.process": [[2, 1, 1, "", "filter_sv_by_clone"], [2, 1, 1, "", "merge_cn_segments"], [2, 1, 1, "", "update_blocks_and_reset_prev"]], "ontmont.utils": [[2, 1, 1, "", "enumerate_breakpoints"], [2, 1, 1, "", "extract_split_alignments"], [2, 1, 1, "", "filter_breakpoints_at_contig_ends"], [2, 1, 1, "", "filter_sv_with_breakpoint_at_contig_ends"], [2, 1, 1, "", "get_chromosomes_to_process"], [2, 1, 1, "", "get_secondaries"], [2, 1, 1, "", "is_breakpoints_not_sorted"], [2, 1, 1, "", "make_split_read_table"], [2, 1, 1, "", "remove_duplicates_from_tra_table"], [2, 1, 1, "", "reverse_complement"], [2, 1, 1, "", "shuffle_seq"]]}, "objtypes": {"0": "py:module", "1": "py:function", "2": "py:class", "3": "py:attribute", "4": "py:method"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"], "2": ["py", "class", "Python class"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "method", "Python method"]}, "titleterms": {"welcom": 0, "ontmont": [0, 1, 2], "": 0, "document": 0, "content": [0, 2], "indic": 0, "tabl": 0, "packag": 2, "submodul": 2, "bundl": 2, "modul": 2, "cli": 2, "collect": 2, "datatyp": 2, "ir": 2, "plot": 2, "process": 2, "util": 2}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"Welcome to ontmont\u2019s documentation!": [[0, "welcome-to-ontmont-s-documentation"]], "Contents:": [[0, null]], "Indices and tables": [[0, "indices-and-tables"]], "ontmont": [[1, "ontmont"]], "ontmont package": [[2, "ontmont-package"]], "Submodules": [[2, "submodules"]], "ontmont.bundle module": [[2, "module-ontmont.bundle"]], "ontmont.cli module": [[2, "module-ontmont.cli"]], "ontmont.collect module": [[2, "module-ontmont.collect"]], "ontmont.datatypes module": [[2, "module-ontmont.datatypes"]], "ontmont.irs module": [[2, "module-ontmont.irs"]], "ontmont.plot module": [[2, "module-ontmont.plot"]], "ontmont.process module": [[2, "module-ontmont.process"]], "ontmont.utils module": [[2, "module-ontmont.utils"]], "Module contents": [[2, "module-ontmont"]]}, "indexentries": {"breakpoint (class in ontmont.datatypes)": [[2, "ontmont.datatypes.Breakpoint"]], "breakpointchain (class in ontmont.datatypes)": [[2, "ontmont.datatypes.BreakpointChain"]], "breakpointpair (class in ontmont.datatypes)": [[2, "ontmont.datatypes.BreakpointPair"]], "sample (class in ontmont.datatypes)": [[2, "ontmont.datatypes.Sample"]], "segments (class in ontmont.datatypes)": [[2, "ontmont.datatypes.Segments"]], "splitalignment (class in ontmont.datatypes)": [[2, "ontmont.datatypes.SplitAlignment"]], "transitions (class in ontmont.datatypes)": [[2, "ontmont.datatypes.Transitions"]], "add_gene_annotations() (in module ontmont.plot)": [[2, "ontmont.plot.add_gene_annotations"]], "add_sv_legend_to_axes() (in module ontmont.plot)": [[2, "ontmont.plot.add_sv_legend_to_axes"]], "align_two_sequences() (in module ontmont.irs)": [[2, "ontmont.irs.align_two_sequences"]], "calc_distance_score() (in module ontmont.irs)": [[2, "ontmont.irs.calc_distance_score"]], "calc_pval_bilateral_ir() (in module ontmont.irs)": [[2, "ontmont.irs.calc_pval_bilateral_ir"]], "calc_pval_holliday() (in module ontmont.irs)": [[2, "ontmont.irs.calc_pval_holliday"]], "calc_pval_onesided_ir() (in module ontmont.irs)": [[2, "ontmont.irs.calc_pval_onesided_ir"]], "calc_pval_segmental_ir() (in module ontmont.irs)": [[2, "ontmont.irs.calc_pval_segmental_ir"]], "chroms (ontmont.datatypes.breakpoint attribute)": [[2, "ontmont.datatypes.Breakpoint.chroms"]], "convert_tupleproxy_to_pyranges() (in module ontmont.plot)": [[2, "ontmont.plot.convert_tupleproxy_to_pyranges"]], "enumerate_breakpoints() (in module ontmont.utils)": [[2, "ontmont.utils.enumerate_breakpoints"]], "extract_cigar_field() (ontmont.datatypes.splitalignment method)": [[2, "ontmont.datatypes.SplitAlignment.extract_cigar_field"]], "extract_split_alignments() (in module ontmont.collect)": [[2, "ontmont.collect.extract_split_alignments"]], "extract_split_alignments() (in module ontmont.utils)": [[2, "ontmont.utils.extract_split_alignments"]], "extract_vaf_from_annotated_sv_table() (in module ontmont.plot)": [[2, "ontmont.plot.extract_vaf_from_annotated_sv_table"]], "filter_breakpoints_at_contig_ends() (in module ontmont.utils)": [[2, "ontmont.utils.filter_breakpoints_at_contig_ends"]], "filter_sv_by_clone() (in module ontmont.process)": [[2, "ontmont.process.filter_sv_by_clone"]], "filter_sv_with_breakpoint_at_contig_ends() (in module ontmont.utils)": [[2, "ontmont.utils.filter_sv_with_breakpoint_at_contig_ends"]], "find_presence_of_matching_sv() (in module ontmont.collect)": [[2, "ontmont.collect.find_presence_of_matching_sv"]], "fix_coordinate_xticklabels() (in module ontmont.plot)": [[2, "ontmont.plot.fix_coordinate_xticklabels"]], "fix_lower_support_coordinates() (in module ontmont.collect)": [[2, "ontmont.collect.fix_lower_support_coordinates"]], "fix_yticks_with_integers() (in module ontmont.plot)": [[2, "ontmont.plot.fix_yticks_with_integers"]], "get_best_holliday_junctions() (in module ontmont.irs)": [[2, "ontmont.irs.get_best_holliday_junctions"]], "get_best_ir_within_breakpoints() (in module ontmont.irs)": [[2, "ontmont.irs.get_best_ir_within_breakpoints"]], "get_best_ir_within_segment() (in module ontmont.irs)": [[2, "ontmont.irs.get_best_ir_within_segment"]], "get_best_onesided_ir() (in module ontmont.irs)": [[2, "ontmont.irs.get_best_onesided_ir"]], "get_breakpoint_pair_seq_data() (in module ontmont.irs)": [[2, "ontmont.irs.get_breakpoint_pair_seq_data"]], "get_breakpoint_seqs() (in module ontmont.datatypes)": [[2, "ontmont.datatypes.get_breakpoint_seqs"]], "get_breakpoint_seqs() (ontmont.datatypes.breakpoint method)": [[2, "ontmont.datatypes.Breakpoint.get_breakpoint_seqs"]], "get_breakpoint_support_from_bundle() (in module ontmont.collect)": [[2, "ontmont.collect.get_breakpoint_support_from_bundle"]], "get_chrom_xlims() (in module ontmont.plot)": [[2, "ontmont.plot.get_chrom_xlims"]], "get_chromosomes_from_bundle() (in module ontmont.plot)": [[2, "ontmont.plot.get_chromosomes_from_bundle"]], "get_chromosomes_to_process() (in module ontmont.utils)": [[2, "ontmont.utils.get_chromosomes_to_process"]], "get_cigar_tuples() (ontmont.datatypes.splitalignment static method)": [[2, "ontmont.datatypes.SplitAlignment.get_cigar_tuples"]], "get_cn_blocks_for_read_xlim() (in module ontmont.plot)": [[2, "ontmont.plot.get_cn_blocks_for_read_xlim"]], "get_edgecolor() (in module ontmont.plot)": [[2, "ontmont.plot.get_edgecolor"]], "get_gene_repr_exons() (in module ontmont.plot)": [[2, "ontmont.plot.get_gene_repr_exons"]], "get_list() (ontmont.datatypes.segments method)": [[2, "ontmont.datatypes.Segments.get_list"]], "get_list() (ontmont.datatypes.transitions method)": [[2, "ontmont.datatypes.Transitions.get_list"]], "get_normalized_sv() (in module ontmont.collect)": [[2, "ontmont.collect.get_normalized_sv"]], "get_onesided_ir() (in module ontmont.irs)": [[2, "ontmont.irs.get_onesided_ir"]], "get_repr_transcript_id() (in module ontmont.plot)": [[2, "ontmont.plot.get_repr_transcript_id"]], "get_secondaries() (in module ontmont.utils)": [[2, "ontmont.utils.get_secondaries"]], "get_segments() (ontmont.datatypes.breakpointchain method)": [[2, "ontmont.datatypes.BreakpointChain.get_segments"]], "get_sv_table_chroms() (in module ontmont.plot)": [[2, "ontmont.plot.get_sv_table_chroms"]], "get_svtype() (in module ontmont.collect)": [[2, "ontmont.collect.get_svtype"]], "get_transcript_exons() (in module ontmont.plot)": [[2, "ontmont.plot.get_transcript_exons"]], "get_transitions() (ontmont.datatypes.breakpointchain method)": [[2, "ontmont.datatypes.BreakpointChain.get_transitions"]], "get_unique_sv_with_support() (in module ontmont.plot)": [[2, "ontmont.plot.get_unique_sv_with_support"]], "is_breakpoints_not_sorted() (in module ontmont.utils)": [[2, "ontmont.utils.is_breakpoints_not_sorted"]], "is_holliday_junction() (in module ontmont.irs)": [[2, "ontmont.irs.is_holliday_junction"]], "is_overlap() (in module ontmont.plot)": [[2, "ontmont.plot.is_overlap"]], "long_samples (ontmont.datatypes.sample attribute)": [[2, "ontmont.datatypes.Sample.long_samples"]], "make_axes_for_breakpoints() (in module ontmont.plot)": [[2, "ontmont.plot.make_axes_for_breakpoints"]], "make_brk_supports() (in module ontmont.bundle)": [[2, "ontmont.bundle.make_brk_supports"]], "make_brk_table() (in module ontmont.bundle)": [[2, "ontmont.bundle.make_brk_table"]], "make_brks_bundle() (in module ontmont.bundle)": [[2, "ontmont.bundle.make_brks_bundle"]], "make_brks_bundle() (in module ontmont.collect)": [[2, "ontmont.collect.make_brks_bundle"]], "make_seg_table() (in module ontmont.bundle)": [[2, "ontmont.bundle.make_seg_table"]], "make_spline_coordinates() (in module ontmont.plot)": [[2, "ontmont.plot.make_spline_coordinates"]], "make_split_read_table() (in module ontmont.utils)": [[2, "ontmont.utils.make_split_read_table"]], "make_tra_table() (in module ontmont.bundle)": [[2, "ontmont.bundle.make_tra_table"]], "make_tumor_sv_table() (in module ontmont.collect)": [[2, "ontmont.collect.make_tumor_sv_table"]], "map_similar_coordinate_to_higher_rank() (in module ontmont.collect)": [[2, "ontmont.collect.map_similar_coordinate_to_higher_rank"]], "merge_cn_segments() (in module ontmont.process)": [[2, "ontmont.process.merge_cn_segments"]], "module": [[2, "module-ontmont"], [2, "module-ontmont.bundle"], [2, "module-ontmont.cli"], [2, "module-ontmont.collect"], [2, "module-ontmont.datatypes"], [2, "module-ontmont.irs"], [2, "module-ontmont.plot"], [2, "module-ontmont.process"], [2, "module-ontmont.utils"]], "normalize_sv_table() (in module ontmont.collect)": [[2, "ontmont.collect.normalize_sv_table"]], "ontmont": [[2, "module-ontmont"]], "ontmont.bundle": [[2, "module-ontmont.bundle"]], "ontmont.cli": [[2, "module-ontmont.cli"]], "ontmont.collect": [[2, "module-ontmont.collect"]], "ontmont.datatypes": [[2, "module-ontmont.datatypes"]], "ontmont.irs": [[2, "module-ontmont.irs"]], "ontmont.plot": [[2, "module-ontmont.plot"]], "ontmont.process": [[2, "module-ontmont.process"]], "ontmont.utils": [[2, "module-ontmont.utils"]], "overlaps() (in module ontmont.irs)": [[2, "ontmont.irs.overlaps"]], "parse_gtf_region() (in module ontmont.plot)": [[2, "ontmont.plot.parse_gtf_region"]], "plot_cn_on_axes() (in module ontmont.plot)": [[2, "ontmont.plot.plot_cn_on_axes"]], "plot_gene_annotations() (in module ontmont.plot)": [[2, "ontmont.plot.plot_gene_annotations"]], "plot_jabba_cn() (in module ontmont.plot)": [[2, "ontmont.plot.plot_jabba_cn"]], "plot_quasijabba() (in module ontmont.plot)": [[2, "ontmont.plot.plot_quasijabba"]], "plot_segments_on_axes() (in module ontmont.plot)": [[2, "ontmont.plot.plot_segments_on_axes"]], "plot_sv_lines_on_cn() (in module ontmont.plot)": [[2, "ontmont.plot.plot_sv_lines_on_cn"]], "plot_svs_to_cn_segments() (in module ontmont.plot)": [[2, "ontmont.plot.plot_svs_to_cn_segments"]], "plot_transition_on_axes() (in module ontmont.plot)": [[2, "ontmont.plot.plot_transition_on_axes"]], "plot_vaf_cn_sv_for_reads() (in module ontmont.plot)": [[2, "ontmont.plot.plot_vaf_cn_sv_for_reads"]], "plot_vaf_on_axes() (in module ontmont.plot)": [[2, "ontmont.plot.plot_vaf_on_axes"]], "pull_breakpoints_from_bam_files() (in module ontmont.collect)": [[2, "ontmont.collect.pull_breakpoints_from_bam_files"]], "pull_breakpoints_from_reads_in_sv_regions() (in module ontmont.collect)": [[2, "ontmont.collect.pull_breakpoints_from_reads_in_sv_regions"]], "pull_sv_supporting_reads_from_bundle() (in module ontmont.collect)": [[2, "ontmont.collect.pull_sv_supporting_reads_from_bundle"]], "remove_duplicates_from_tra_table() (in module ontmont.utils)": [[2, "ontmont.utils.remove_duplicates_from_tra_table"]], "reverse_complement() (in module ontmont.utils)": [[2, "ontmont.utils.reverse_complement"]], "sample2group (ontmont.datatypes.sample attribute)": [[2, "ontmont.datatypes.Sample.sample2group"]], "sample2short (ontmont.datatypes.sample attribute)": [[2, "ontmont.datatypes.Sample.sample2short"]], "shuffle_seq() (in module ontmont.utils)": [[2, "ontmont.utils.shuffle_seq"]], "update_blocks_and_reset_prev() (in module ontmont.process)": [[2, "ontmont.process.update_blocks_and_reset_prev"]]}})